### PR TITLE
ENYO-6009: ui/ToggleItem value prop is not sent to onToggle

### PR DESCRIPTION
**Pull request by sugardave**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Components that use `ui/Toggleable` HOC do not receive the `value` prop in their toggle handlers as detailed in the docs for `ui/ToggleItem`.  This patch addresses that.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
`ui/Toggleable` HOC receives a new `eventProps` config option and `ui/ToggleItemDecorator` now configures `{toggleProp: 'onTap', eventProps: ['value']}` when it is composing.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
There is something odd about the way `ui/Toggleable` overwrites `ui/ToggleItem`'s `onToggle` handler but doesn't forward the event properly that I haven't finished puzzling out.

### Links
[//]: # (Related issues, references)
ENYO-6009

### CODE REVIEW

File: packages/ui/ToggleItem/ToggleItem.js [hunk 1]
```diff
@@ -11,6 +11,7 @@
  * @exports ToggleItemDecorator
  */
 
+import {adaptEvent, handle, forward} from '@enact/core/handle';
```
Comment (Djspaceg) [inline packages/ui/ToggleItem/ToggleItem.js]: This is no longer needed, right?

File: packages/ui/Toggleable/Toggleable.js [hunk 2]
```diff
@@ -204,7 +212,7 @@ const ToggleableHOC = hoc(defaultConfig, (config, Wrapped) => {
 
 		handle = handle.bind(this)
 
-		forwardWithState = (evName) => (ev, props) => forward(evName, {[prop]: !this.state.active}, props)
+		forwardWithState = (evName) => (ev, props) => forward(evName, {...pick(eventProps, props), [prop]: !this.state.active}, props)
```
Comment (Djspaceg) [inline packages/ui/Toggleable/Toggleable.js]: If `eventProps` is somehow undefined/null, does `pick` fall over? Probably an over-engineered edge case, but I'm curious.

File: packages/ui/Toggleable/Toggleable.js [hunk 3]
```diff
@@ -63,6 +64,13 @@ const defaultConfig = {
 	 */
 	deactivate: null,
 
+	/**
+	 * Configures additional props to attach to the event that is sent when toggled.
+	 *
+	 * @type {String[]}
```
Comment (webOS101) [inline packages/ui/Toggleable/Toggleable.js]: Should we add a `@default` here?  I think, probably, yes.  I can do it.

Comment (webOS101) [reply packages/ui/Toggleable/Toggleable.js]: Also, need a `@memberof` here to prevent parse errors.

**Pull request merged**